src/messageSigning.c

```diff
@@ -43,11 +43,9 @@ void getTxWitness(bip44_path_t* pathSpec,
 		TRY {
 			derivePrivateKey(pathSpec, &chainCode, &privateKey);
```

Comment:
Imagine having a `noinline` function wrapping `derivePrivateKey` (which would more accurately be called `deriveExtendedPrivateKey`) and throwing away the chain code which we never need for signing. I think it will save some stack space. Might be handy in the future.

Reply:
IMHO this is enough to do when we face actual stack space shortage while signing and do a proper analysis of the root cause should that be the case. I'd rather not trade-off complexity for memory usage if we don't urgently need it